cpp/src/scalar/scalar_factories.cpp

```diff
@@ -165,4 +167,25 @@ std::unique_ptr<scalar> make_default_constructed_scalar(data_type type,
   return type_dispatcher(type, default_scalar_functor{}, stream, mr);
 }
 
+std::unique_ptr<scalar> make_empty_scalar_like(column_view const& column,
+                                               rmm::cuda_stream_view stream,
+                                               rmm::mr::device_memory_resource* mr)
+{
+  std::unique_ptr<scalar> result;
+  switch (column.type().id()) {
+    case type_id::LIST:
+      result = make_list_scalar(empty_like(column)->view(), stream, mr);
+      result->set_valid_async(false, stream);
+      break;
+    case type_id::STRUCT:
+      // Struct scalar inputs must have exactly 1 row.
```

Comment:
Should have "at least one row", I think...

Reply:
https://github.com/rapidsai/cudf/blob/branch-21.10/cpp/src/scalar/scalar.cpp#L567-L568

Reply:
If so, you don't have to have your own `CUDF_EXPECT` because `rows == 1` will be handled in the scalar constructor.

Reply:
Wait, then you should use `make_struct_scalar` too :|